packages/core/react/src/useLocalStorage.ts

```diff
@@ -12,23 +12,22 @@ export function useLocalStorage<T>(key: string, defaultState: T): [T, (newValue:
         return defaultState;
     });
 
-    const setLocalStorage = useCallback(
-        (newValue: T) => {
-            if (newValue === value) return;
-            setValue(newValue);
-
-            try {
-                if (newValue === null) {
-                    localStorage.removeItem(key);
-                } else {
-                    localStorage.setItem(key, JSON.stringify(newValue));
-                }
-            } catch (error) {
-                console.error(error);
+    const isFirstRender = useRef(true);
+    useEffect(() => {
+        if (isFirstRender.current === true) {
+            isFirstRender.current = false;
+            return;
+        }
+        try {
+            if (state[0] === null) {
+                localStorage.removeItem(key);
+            } else {
+                localStorage.setItem(key, JSON.stringify(state[0]));
             }
-        },
-        [value, setValue, key]
-    );
+        } catch (error) {
+            console.error(error);
+        }
+    }, [state[0]]);
```

Comment:
This makes a ton of sense. Using `useEffect` to respond to changes in the state is conceptually cleaner than hybridizing the state setter.

Why do you use `state[0]`?

Reply:
I wanted to pass the (state, stateSetter) tuple _straight_ through, rather than to unwrap it by destructuring only to re-wrap back up the same way but in a new array.

Reply:
Right, I just didn't know if there was a meaningful perf benefit from it?

Reply:
Miniscule. Just wanted to avoid creating and GC-ing yet-another-array.